light-node/src/commands/initialize.rs

```diff
@@ -51,7 +50,22 @@ impl Runnable for InitCmd {
 
         let io = ProdIo::new(peer_map, Some(app_cfg.rpc_config.request_timeout));
 
-        initialize_subjectively(self.height, subjective_header_hash, &lc, &io);
+        let lock = APPLICATION.read();
+
+        let factory: &dyn LightStoreFactory = lock
+            .state()
+            .components
+            .get_downcast_ref::<ProdLightStoreFactory>()
```

Comment:
I don't believe we need the flexibility afforded by the `LightStoreFactory` trait/component, a plain method which converts a `LightStoreConfig` to a `impl LightStore` would likely do the job as well.

This line is also a bit weird because we are always casting to a concrete type, which defeats the purpose of the trait.

Reply:
Changed it to a function.